<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Провести-дисперсионный-анализ-для-определения-того,-есть-ли-различия-среднего-роста-среди-взрослых-футболистов,-хоккеистов-и-штангистов.-Даны-значения-роста-в-трех-группах-случайно-выбранных-спортсменов:" data-toc-modified-id="Провести-дисперсионный-анализ-для-определения-того,-есть-ли-различия-среднего-роста-среди-взрослых-футболистов,-хоккеистов-и-штангистов.-Даны-значения-роста-в-трех-группах-случайно-выбранных-спортсменов:-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов:</a></span><ul class="toc-item"><li><span><a href="#Вывод:" data-toc-modified-id="Вывод:-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Вывод:</a></span></li></ul></li></ul></div>

### Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов:

Футболисты: 173, 175, 180, 178, 177, 185, 183, 182.

Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180.

Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [23]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import pandas as pd
import numpy as np
from scipy.stats import bartlett
from scipy.stats import shapiro

In [24]:
a1 = [173, 175, 180, 178, 177, 185, 183, 182]
a2 = [177, 179, 180, 188, 177, 172, 171, 184, 180]
a3 = [172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170]

In [26]:
bartlett(a1,a2,a3)

BartlettResult(statistic=0.4640521043406442, pvalue=0.7929254656083131)

Тест Бартлетта говорит о том, что нужно оставить Н0, дисперсии статестически равны.

In [27]:
d1 = shapiro(a1)
d2 = shapiro(a2)
d3 = shapiro(a3)
d1,d2,d3

(ShapiroResult(statistic=0.9775082468986511, pvalue=0.9495404362678528),
 ShapiroResult(statistic=0.9579196572303772, pvalue=0.7763139009475708),
 ShapiroResult(statistic=0.9386808276176453, pvalue=0.5051165223121643))

Во всех выборках p-value больше 5%, значит все выборки нормально распределены.

In [19]:
df = pd.DataFrame({'score': [173, 175, 180, 178, 177, 185, 183, 182,
                            177, 179, 180, 188, 177, 172, 171, 184, 180,
                            172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170],
                  'group': np.repeat(['football','hockey','weightlifter'], repeats=[8, 9, 11])})

In [20]:
tukey = pairwise_tukeyhsd(endog=df['score'],
                         groups=df['group'],
                         alpha=0.05)

In [21]:
print(tukey)

     Multiple Comparison of Means - Tukey HSD, FWER=0.05     
 group1     group2    meandiff p-adj   lower    upper  reject
-------------------------------------------------------------
football       hockey  -0.4583  0.979  -6.2732  5.3566  False
football weightlifter  -6.3977 0.0219 -11.9583 -0.8372   True
  hockey weightlifter  -5.9394 0.0284 -11.3181 -0.5607   True
-------------------------------------------------------------


#### Вывод: 
Тест Тьюки показывает значимые различия между группой штангистов(`weightlifter`) и группами футболистов и хоккеистов. Уровень значимости p-value при сравнении средних данных групп 2.19% и 2.84% соответственно. Так же можно сказать, что средний рост штангистов статистически значимо ниже остальных выбранных спортсменов.

Статистически значимых различий между футболистами и хоккеистами не обнаружено, p-value при сравнении данных групп 97.9%